src/merge-analysis/into-classes.js

```diff
@@ -91,8 +92,15 @@ function intoClasses(classes, members) {
 
   // Generate the class code
   for (const pair of values(pairs)) {
+    // This can happen when your documented members are all marked as private OR something is not documented.
     if (pair.members.length === 0) {
-      info(pair.class, 'Class', `path ${pair.path} has no defined members`);
+      info(pair.class, 'Class', `path ${pair.path} has no defined members, nothing will be documented.`);
+    }
+
+    // Do not add private classes
+    if (pair.class.comment.includes("@private")) {
```

Comment:
We can be more specific here.
1.  Only output the error if the comment contains `@private` and is not a function; meaning, if there is a `@name` or `namespace` tag, or no `@static` nor `@constructor` tag. "Private class"
2.  Output an error if there is both  `@static`/`@constructor` and a `@name*` tag. "method documented like a class"
3.  Output an error if there is a `@static` or `@constructor` tag "function name looks like a class name"

Reply:
You want multiple distinct error message here? To get parsed at all it the code block will have to have had a `@name` tag

Reply:
In that case, check for static, constructor, instance, function, return, callback, etc, if any tag which does not belong on a class is found, output "Tag @XXX found on a class definition, did you mean it to be a function?".
Unfortunately there is nothing which will find that CIQ.Share.FullChart2PNG is not really a class or a constructor, but rather just a function poorly named.  But if you put a `@static` tag on it, I'd like it to  catch and output the above error.

Reply:
One message per doc block is enough.

Reply:
If the idea is to check class definitions for tags that do not belong there then we should probably be doing that earlier not at the very end. I'll see what I can do.

Reply:
I think instead the better place to do any checking like this would be when we're already parsing comments to make sure there's no 'mutually exclusive' tags.